quickwit-metastore/src/metastore/index_metadata.rs

```diff
@@ -0,0 +1,414 @@
+        builder.tag_fields = self.doc_mapping.tag_fields.iter().cloned().collect();
+        builder.store_source = self.doc_mapping.store_source;
+        Ok(Arc::new(builder.build()?))
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub(crate) struct UnversionedIndexMetadata {
+    pub index_id: String,
+    pub index_uri: String,
+    pub index_config: DefaultDocMapper,
+    pub checkpoint: Checkpoint,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(tag = "version")]
+pub(crate) enum VersionedIndexMetadata {
+    #[serde(rename = "1")]
+    V1(IndexMetadataV1),
+    #[serde(rename = "0")]
+    V0(IndexMetadataV0),
+    #[serde(rename = "unversioned")]
+    Unversioned(UnversionedIndexMetadata),
+}
+
+impl From<IndexMetadataV1> for VersionedIndexMetadata {
+    fn from(index_metadata: IndexMetadataV1) -> Self {
+        VersionedIndexMetadata::V1(index_metadata)
+    }
+}
+
+impl From<VersionedIndexMetadata> for IndexMetadata {
+    fn from(index_metadata: VersionedIndexMetadata) -> Self {
+        match index_metadata {
+            VersionedIndexMetadata::Unversioned(unversioned) => unversioned.into(),
+            VersionedIndexMetadata::V0(v0) => v0.into(),
+            VersionedIndexMetadata::V1(v1) => v1.into(),
+        }
+    }
+}
+
+impl From<IndexMetadata> for IndexMetadataV1 {
+    fn from(index_metadata: IndexMetadata) -> Self {
+        Self {
+            index_id: index_metadata.index_id,
+            index_uri: index_metadata.index_uri,
+            source_checkpoints: index_metadata.source_checkpoints,
+            doc_mapping: index_metadata.doc_mapping,
+            indexing_settings: index_metadata.indexing_settings,
+            search_settings: index_metadata.search_settings,
+            sources: index_metadata.sources,
+            create_timestamp: index_metadata.create_timestamp,
+            update_timestamp: index_metadata.update_timestamp,
+        }
+    }
+}
+
+impl From<IndexMetadataV1> for IndexMetadata {
+    fn from(index_metadata: IndexMetadataV1) -> Self {
+        Self {
+            index_id: index_metadata.index_id,
+            index_uri: index_metadata.index_uri,
+            source_checkpoints: index_metadata.source_checkpoints,
+            doc_mapping: index_metadata.doc_mapping,
+            indexing_settings: index_metadata.indexing_settings,
+            search_settings: index_metadata.search_settings,
+            sources: index_metadata.sources,
+            create_timestamp: index_metadata.create_timestamp,
+            update_timestamp: index_metadata.update_timestamp,
+        }
+    }
+}
+
+impl From<IndexMetadata> for VersionedIndexMetadata {
+    fn from(index_metadata: IndexMetadata) -> Self {
+        VersionedIndexMetadata::V1(index_metadata.into())
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub(crate) struct IndexMetadataV0 {
+    pub index_id: String,
+    pub index_uri: String,
+    #[serde(default)]
```

Comment:
We would have to remove those two lines.

Reply:
discussed offline. we don't want a default.

Reply:
Done.